### DISCUSSION TITLE
Scraping Data Fields From Dynamic Website

### DISCUSSION THREAD

**Discussion by alpha-code2019** [AutomaApp/automa#515] [Q&A] [answered]
**Hi, I think that Automa is a really great app - thank you!**

I have the following question:

On many websites, the available/possible data fields are only shown to the website visitor if the data field has a value. If there is no value, then the data field is not shown.

That mixes scraped data up, a lot.

Is there a way to tell Automa that 
If **Data Field 1** exists, then scrape the available data value and save it as a variable for later use (add to a Google table for instance)

Example
**Data Field 1:**     Data Field 1 Value
**Data Field 2:**     Data Field 2 Value
**Data Field 3:**     Data Field 3 Value

**Data Field 4:**     Has no value and therefore this data field is not shown on that specific webpage, but it could be that it has a value on another webpage, and then it needs to be scraped.

As a real-life example, here are two webpage URLs that contain slightly different data fields:

https://www.bizbuysell.com/Business-Opportunity/retail-flower-and-garden-center-for-sale-very-profitable/1938020/?d=L2Zsb3JpZGEtYnVzaW5lc3Nlcy1mb3Itc2FsZS8/cT1hejFIWVhKa1pXNGdRMlZ1ZEdWeUpteDBQVE13TERRd0xEZ3c=

https://www.bizbuysell.com/Business-Opportunity/retail-garden-center-turf-grass-sales/1969679/?d=L2Zsb3JpZGEtYnVzaW5lc3Nlcy1mb3Itc2FsZS8/cT1hejFIWVhKa1pXNGdRMlZ1ZEdWeUpteDBQVE13TERRd0xEZ3c=

I hope for a solution! Thank you for looking into this problem

**Comment by Kholid060** [comment, marked as answer]
In that case, I think you can use the conditions block or the element exists block. You can use it to check whether the **Data Field 4** exists or not. When the data field exists, you can tell Automa to scrape it, and if it's not, execute the fallback block.

![image](https://user-images.githubusercontent.com/22908993/170901672-beed0cc3-c06a-4986-bfc0-77ce719d5f6d.png)

**Comment by alpha-code2019** [comment]
Thank you, Kholid. I will try and report if it works.
